src/mysql_router_helpers.py

```diff
@@ -132,15 +97,13 @@ def bootstrap_and_start_mysql_router(
         # https://dev.mysql.com/doc/refman/8.0/en/caching-sha2-pluggable-authentication.html)
         bootstrap_mysqlrouter_command = [
             "sudo",
-            "/usr/bin/mysqlrouter",
+            CHARMED_MYSQL_ROUTER,
             "--user",
-            MYSQL_ROUTER_USER,
+            SNAP_DAEMON_USER,
             "--name",
             name,
             "--bootstrap",
             f"{user}:{password}@{db_host}",
-            "--directory",
-            f"{MYSQL_HOME_DIRECTORY}/{name}",
```

Comment:
Since there is a 1:1 relation between mysqlrouter and the database (at least in the VM charm), I envisioned multiple mysqlrouter charms deployed to the same machine achieve parallelism using parallel installs with the `charmed-mysql` snap itself.

However, may be we need `--directory` to support relations with multiple applications in the k8s charm. 

Thoughts?

Reply:
Yeah, I think that that makes sense. I wonder if the `--name` flag becomes obsolete in that case in this PR?

Reply:
I see the use-cases to install 2+ mysql-router as subordinate charms in the same app.
e.g. you have two independent clusters and app need access there.
@shayancanonical I think we discussed this yesterday, did we misunderstand each other?

Reply:
based on our discussion during the previous database office hours: we will only ever need the mysqlrouter process to connect to one database cluster at a time. however, the application can relate to mysqlrouter multiple times to request different databases in the same database cluster. since the current operator does not correctly support multiple relations with the application, i suggest we leave this PR as is. however, @carlcsaposs-canonical's refactor of the charm will address this concern

to install 2 different mysqlrouter applications as subordinate charms on the same machine, we need to utilize the experimental feature `parallel installs` for snaps. however, the details there are still murky and hard to project (the changes to the snap charm library in operator-libs-linux are not immediately clear)